app/utils/transactionHelpers.ts

```diff
@@ -705,3 +688,36 @@ export function isOutgoingTransaction(
 ) {
     return transaction.fromAddress === accountAddress;
 }
+
+/**
+ * Determine whether the transaction affects unshielded balance.
+ */
+export function isUnshieldedBalanceTransaction(
+    transaction: TransferTransaction,
+    currentAddress: string
+) {
+    return !(
+        [
+            TransactionKindString.EncryptedAmountTransfer,
+            TransactionKindString.EncryptedAmountTransferWithMemo,
+        ].includes(transaction.transactionKind) &&
+        transaction.fromAddress !== currentAddress
+    );
+}
+
+/**
+ * Determine whether the transaction affects shielded balance.
+ */
+export function isShieldedBalanceTransaction(
+    transaction: Partial<TransferTransaction>
```

Comment:
I know you just moved this method, but `Partial<TransferTransaction>` allows for `transaction.transactionKind` to not be set. Perhaps you can improve it so that we do not allow it to be optional here.

Reply:
We don't actually need it to be partial, it turns out.